src/MLIR/AST.hs

```diff
@@ -239,6 +239,21 @@ instance FromAST Location Native.Location where
         Native.getFileLineColLocation ctx fileStrRef cline ccol
           where cline = fromIntegral line
                 ccol = fromIntegral col
+    FusedLocation locLocations locMetadata -> do
+      metadata <- case (locMetadata) of
+        -- TODO: Consider factoring out to convenience function.
+        Nothing -> [C.exp| MlirAttribute { mlirAttributeGetNull() } |]
+        Just l -> fromAST ctx env l
+      evalContT $ do
+        -- TODO: This limits the creation of FusedLocation to via AST, this is
+        -- contrary to the other Location which have getters in the Native.hs.
+        -- Revise this to make this more uniform.
```

Comment:
I think this is fine. It's good to try to keep the `Native` part self-contained, but this works too.